### PR TITLE
[Paddle]Add chapter_optimization

### PR DESCRIPTION
**Pull request by tngt**

### CODE REVIEW

**Comment by w5688414** [comment]
chapter chapter_optimization is ready, can you review this chapter? @cheungdaven  thanks

File: chapter_optimization/lr-scheduler.md [hunk 1]
```diff
@@ -357,6 +444,21 @@ scheduler = MultiFactorScheduler(step=[15, 30], factor=0.5, base_lr=0.5)
 d2l.plot(d2l.arange(num_epochs), [scheduler(t) for t in range(num_epochs)])
 ```
 
+```{.python .input}
+#@tab paddle
+net = net_fn()
+scheduler =paddle.optimizer.lr.MultiStepDecay(learning_rate = 0.5, milestones = [15,30], gamma=0.5)
```
Comment (cheungdaven) [inline chapter_optimization/lr-scheduler.md:450]: learning_rate=0.5, milestones=[15, 30], gamma=0.5

File: chapter_optimization/lr-scheduler.md [hunk 2]
```diff
@@ -357,6 +444,21 @@ scheduler = MultiFactorScheduler(step=[15, 30], factor=0.5, base_lr=0.5)
...
+trainer = paddle.optimizer.SGD(learning_rate = scheduler, parameters=net.parameters())
```
Comment (cheungdaven) [inline chapter_optimization/lr-scheduler.md:451]: learning_rate=scheduler,

File: chapter_optimization/minibatch-sgd.md [hunk 3]
```diff
@@ -430,6 +513,34 @@ def train_ch11(trainer_fn, states, hyperparams, data_iter,
     return timer.cumsum(), animator.Y[0]
 ```
 
+```{.python .input}
+#@tab paddle
+#@save
+def train_ch11(trainer_fn, states, hyperparams, data_iter,
+               feature_dim, num_epochs=2):
+    # 初始化模型
+    w = d2l.tensor(d2l.normal(mean=0.0, std=0.01, shape=(feature_dim, 1),),stop_gradient=False)
```
Comment (cheungdaven) [inline chapter_optimization/minibatch-sgd.md:522]: shape=(feature_dim, 1)), stop_gradient=False

**Review by cheungdaven — CHANGES_REQUESTED** [review]

File: chapter_optimization/adadelta.md [hunk 4]
```diff
@@ -107,6 +107,32 @@ def adadelta(params, grads, states, hyperparams):
...
 ```
 
+```{.python .input}
+#@tab paddle
+%matplotlib inline
+from d2l import paddle as d2l
+import paddle
+
+def init_adadelta_states(feature_dim):
+    s_w, s_b = d2l.zeros(shape=(feature_dim, 1)), d2l.zeros(shape=(1,))
```
[the remark this answers is not in the document]
Comment (w5688414) [reply chapter_optimization/adadelta.md:117]: Done

File: chapter_optimization/adadelta.md [hunk 5]
```diff
@@ -107,6 +107,32 @@ def adadelta(params, grads, states, hyperparams):
...
 
+```{.python .input}
+#@tab paddle
+%matplotlib inline
+from d2l import paddle as d2l
+import paddle
+
+def init_adadelta_states(feature_dim):
+    s_w, s_b = d2l.zeros(shape=(feature_dim, 1)), d2l.zeros(shape=(1,)) 
+    delta_w, delta_b = d2l.zeros(shape=(feature_dim, 1)), d2l.zeros(shape=(1,))
```
[the remark this answers is not in the document]
Comment (w5688414) [reply chapter_optimization/adadelta.md:118]: Done

File: chapter_optimization/adagrad.md [hunk 6]
```diff
@@ -216,6 +224,25 @@ def adagrad(params, grads, states, hyperparams):
         p[:].assign(p - hyperparams['lr'] * g / tf.math.sqrt(s + eps))
 ```
 
+```{.python .input}
+#@tab paddle
+def init_adagrad_states(feature_dim):
+    s_w = d2l.zeros((feature_dim, 1))
+    s_b = d2l.zeros(shape=(1,))
```
[the remark this answers is not in the document]
Comment (w5688414) [reply chapter_optimization/adagrad.md:231]: Done

File: chapter_optimization/adam.md [hunk 7]
```diff
@@ -131,6 +131,34 @@ def adam(params, grads, states, hyperparams):
...
 ```
 
+```{.python .input}
+#@tab paddle
+%matplotlib inline
+from d2l import paddle as d2l
+import paddle
+
+def init_adam_states(feature_dim):
+    v_w, v_b = d2l.zeros((feature_dim, 1)), d2l.zeros((1,))
```
[the remark this answers is not in the document]
Comment (w5688414) [reply chapter_optimization/adam.md:141]: Done

File: chapter_optimization/adam.md [hunk 8]
```diff
@@ -131,6 +131,34 @@ def adam(params, grads, states, hyperparams):
...
+    s_w, s_b = d2l.zeros((feature_dim, 1)), d2l.zeros((1,))
```
[the remark this answers is not in the document]
Comment (w5688414) [reply chapter_optimization/adam.md:142]: Done

File: chapter_optimization/lr-scheduler.md (hunk 1 above)
Comment (w5688414) [reply chapter_optimization/lr-scheduler.md:450]: Done

File: chapter_optimization/lr-scheduler.md (hunk 2 above)
Comment (w5688414) [reply chapter_optimization/lr-scheduler.md:451]: Done

File: chapter_optimization/minibatch-sgd.md (hunk 3 above)
Comment (w5688414) [reply chapter_optimization/minibatch-sgd.md:522]: Done

**Comment by cheungdaven** [comment]
@w5688414 could please fix the CI issues, thanks!

**Comment by w5688414** [comment]
> @w5688414 could please fix the CI issues, thanks!

I fixed the missing show_list_len_pair_hist issue, it works fine right now, can you review this chapter again?
